src/main/java/com/ericsson/ei/config/ConfigurationLogger.java

```diff
@@ -1,80 +1,85 @@
-                + "spring.mail.host: " + environment.getProperty("spring.mail.host") + "\n"
-                + "spring.mail.port: " + environment.getProperty("spring.mail.port") + "\n"
-                + "spring.mail.username: " + environment.getProperty("spring.mail.username") + "\n"
-                + "spring.mail.properties.mail.smtp.auth: " + environment.getProperty("spring.mail.properties.mail.smtp.auth") + "\n"
-                + "spring.mail.properties.mail.smtp.starttls.enable: " + environment.getProperty("spring.mail.properties.mail.smtp.starttls.enable") + "\n"
-                + "er.url: " + environment.getProperty("er.url") + "\n"
-                + "ldap.enabled: " + environment.getProperty("ldap.enabled") + "\n"
-                + "ldap.server.list: " + environment.getProperty("ldap.server.list") + "\n"
-                + "logging.level.root: " + environment.getProperty("logging.level.root") + "\n"
-                + "logging.level.org.springframework.web: " + environment.getProperty("logging.level.org.springframework.web") + "\n"
-                + "logging.level.com.ericsson.ei: " + environment.getProperty("logging.level.com.ericsson.ei") + "\n"
-                + "\nThese properties are only some of the configurations, more configurations could have been provided.\n");
+        List<MapPropertySource> propertySources = findApplicationProperties();
+        for (MapPropertySource propertySource : propertySources) {
+            String propertiesLogMessage = createPropertiesLogMessage(propertySource);
+            LOGGER.debug(propertiesLogMessage);
+        }
+    }
+
+    private List<MapPropertySource> findApplicationProperties() {
+        List<MapPropertySource> propertySources = new ArrayList<>();
+        for (PropertySource<?> propertySource : ((ConfigurableEnvironment) environment).getPropertySources()) {
+            if (propertySource.getName().contains("applicationConfig")) {
+                propertySources.add((MapPropertySource) propertySource);
+            }
+        }
+        return propertySources;
+    }
+
+    private String createPropertiesLogMessage(MapPropertySource propertySource) {
+        String[] propertyNames = propertySource.getPropertyNames();
+        Arrays.sort(propertyNames);
+        StringBuilder propertiesLogMessage = new StringBuilder();
+        propertiesLogMessage.append(String.format("\n##### %s #####\n", propertySource.getName()));
+        for (String propertyName : propertyNames) {
+            String propertyValue = getPropertyValueAndCheckForSpecialCases(propertyName);
+            propertiesLogMessage.append(String.format("%s: %s\n", propertyName, propertyValue));
+        }
+        return propertiesLogMessage.toString();
+    }
+
+    private String getPropertyValueAndCheckForSpecialCases(String propertyName) {
+        switch (propertyName) {
+        case "spring.data.mongodb.uri":
```

Comment:
Don't we have more special cases? Like ldap.server.list? and rabbitmq.password?

Reply:
I put in cases for rabbitmq.password, spring.mail.password and ldap.server.list passwords.